lib/internal/process.js

```diff
@@ -283,6 +283,53 @@ function setupUncaughtExceptionCapture(exceptionHandlerState) {
   };
 }
 
+function setupCompareVersion() {
+  const {
+    majorVersion,
+    minorVersion,
+    patchVersion,
+  } = process.release;
+  const prereleaseTag = process.release.prereleaseTag.replace(/^-/, '');
+
+  process.release.compareVersion =
+    (major, minor, patch, prerelease = false) => {
+      if (typeof major === 'string')
+        [major, minor, patch] = major.split('.');
+      if (/-/.test(patch))
+        [patch, prerelease] = patch.split('-');
+
+      major = +major;
+      minor = +minor;
+      patch = +patch;
+      if (/^-/.test(prerelease))
+        prerelease = prerelease.slice(1);
+
+      if (major > majorVersion)
+        return 1;
+      if (major < majorVersion)
+        return -1;
+
+      if (minor > minorVersion)
+        return 1;
+      if (minor < minorVersion)
+        return -1;
+
+      if (prereleaseTag && prereleaseTag === prerelease &&
+          patch === patchVersion)
+        return 0;
+      if (prereleaseTag && patch === patchVersion)
```

Comment:
I think the the above two checks could be grouped together under a `if (prereleaseTag)` conditional to avoid checking it twice.